Registry: Document and flatten configuration

Geod24 opened the pull request:
```
The registry configuration was a bit lacking, and this has been fixed.
Additionally, since the whole point of the 'registry' is now to have an 'enabled' boolean,
and then contain the DNS configuration, the two configurations were flattened.
One reason to do so is that we want to move towards the registry being mostly DNS-based
(as opposed to mostly HTTP-based, as currently), if not only DNS-based.
Hence, the registry and the DNS server are now tied together.
The original reason for allowing disabling the DNS server was to not impair
production if a bug was found in the DNS server, but this concern is no more.
```

Geod24 committed:
Registry: Document and flatten configuration

The registry configuration was a bit lacking, and this has been fixed.
Additionally, since the whole point of the 'registry' is now to have an 'enabled' boolean,
and then contain the DNS configuration, the two configurations were flattened.
One reason to do so is that we want to move towards the registry being mostly DNS-based
(as opposed to mostly HTTP-based, as currently), if not only DNS-based.
Hence, the registry and the DNS server are now tied together.
The original reason for allowing disabling the DNS server was to not impair
production if a bug was found in the DNS server, but this concern is no more.

Geod24 committed:
Tests: Pass a valid config to TestNameRegistry

hewison-chris commented on source/agora/node/Config.d:
```diff
@@ -482,6 +412,63 @@ public struct DNSConfig
+                   zone.name, zone.expire, intMaxSecs);
+            ensure(zone.minimum <= intMaxSecs,
+                   "registry.authoritative: Zone '%s' field 'minimum' ({}) should be at most {}",
+                   zone.name, zone.minimum, uintMaxSecs);
```
```suggested
                   zone.name, zone.minimum, intMaxSecs);
```

Geod24 replied:
Good eye! Although the error is actually on the comparison, as `minimum` is unsigned:

> MINIMUM         The unsigned 32 bit minimum TTL field that should be
>                           exported with any RR from this zone.

[Source](https://datatracker.ietf.org/doc/html/rfc1035#section-3.3.13).
Added a commit to fix it.

hewison-chris reviewed (changes requested):
LGTM just a typo

kchulj commented on source/agora/node/Runner.d:
```diff
@@ -130,12 +130,9 @@ public Listeners runNode (Config config)
-            result.tcp ~= listenTCP(config.registry.dns.port, (conn) => conn.runTCPDNSServer(result.registry),
-                config.registry.dns.address);
-        }
+        /* auto dnstask = */ runTask(() => runDNSServer(config.registry, result.registry));
```
Is the `auto dnstask` an implementation for the future ?

Geod24 replied:
It's a bit of a tricky issue with the UDP code.
The TCP code spawns fibers for us and our client code run in it. But the UDP code is a bit more ancient and doesn't do that. Instead, we need to manually run the server and listen in it. Which means we need to run a task.
When the node shuts down, we should interrupt that task. However, interrupt throws an Exception, so we would need to handle that. So I've let it for a future enhancement.

linked0 reviewed (approved):
LGTM

kchulj commented on doc/config.example.yaml:
```diff
@@ -272,28 +272,55 @@ event_handlers:
+      ##
+      ## Those fields are ignored if the server is not authoritative for the zone.
+
+      # Email of the DNS server admnistrator
```
```suggestion
      # Email of the DNS server administrator
```

Geod24 replied:
Fixed

kchulj reviewed (approved):
LGTM

Geod24 committed:
Registry: Fix validation of SOA's minimum field

The error message was showing the right value, but the comparison was wrong.
Minimum is the only unsigned field in this structure.

hewison-chris reviewed (approved):
LGTM
